### PR TITLE
[Docs] 7.4 Updated ML Job Descriptions

### PR DESCRIPTION
**Pull request by benskelker**
Adds @randomuserid updated descriptions of new ML jobs.

Replaces [PR 492](https://github.com/elastic/stack-docs/pull/492), primarily because I screwed up my local repo...

[Preview](http://stack-docs_509.docs-preview.app.elstc.co/guide/en/siem/guide/master/machine-learning.html)

### CODE REVIEW

File: docs/en/siem/machine-learning.asciidoc [hunk 1]
```diff
@@ -21,35 +21,185 @@ image::ml-ui.png[]
...
-and `winlogbeat-*`) via {kib} -> Management -> Index Patterns, the jobs will be
-installed on page load, and will be displayed within the `Anomaly Detection`
-interface.
+If you ship data using https://www.elastic.co/products/beats[Beats] and your environment
+is configured with the appropriate indices (`auditbeat-*`, `filebeat-*`, `packetbeat-*`,
+or `winlogbeat-*`) via {kib} -> Management -> Index Patterns, the jobs will be installed
+on page load, and will be displayed within the `Anomaly Detection` interface.
+
+NOTE: Machine learning jobs look back and analyse two weeks of historical data prior to
+the time they are enabled. After jobs are enabled, they continuously analyse incoming data.
```
Comment (randomuserid) [inline docs/en/siem/machine-learning.asciidoc]: two weeks is the conservative default which can be increased and should be increased when capacity permits - the more data the models longest, they better they get.

File: docs/en/siem/machine-learning.asciidoc [hunk 2]
```diff
@@ -21,35 +21,185 @@ image::ml-ui.png[]
...
++
+Beats required on hosts:
+
+* Auditbeat (Linux)
+* Winlogbeat (Windows)
+windows_anomalous_process_creation:: Identifies unusual parent/child process
+relationships that could indicate malware execution or persistence mechanisms.
++
+Malicious scripts often call on other applications and processes as part of their
+exploit payload. For example, a malicious Office document runs a script that starts
```
Comment (randomuserid) [inline docs/en/siem/machine-learning.asciidoc]: original was "For example, when a malicious Office document runs scripts as part of an exploit payload, Excel or Word may start a script interpreter process, which runs a script that downloads and runs malware."

File: docs/en/siem/machine-learning.asciidoc [hunk 3]
```diff
@@ -21,35 +21,185 @@ image::ml-ui.png[]
...
+an Excel or Word script interpreter process, which, in turn, executes another script
+that downloads and runs malware. Another common scenario is Outlook running an
+unusual process when malware is downloaded in an email.
++
+Monitoring and identifying process relationships is an excellent way of detecting
```
Comment (randomuserid) [inline docs/en/siem/machine-learning.asciidoc]: identifying anomalous process relationships

File: docs/en/siem/machine-learning.asciidoc [hunk 4]
```diff
@@ -21,35 +21,185 @@ image::ml-ui.png[]
...
+new and emerging malware that is not yet recognized by anti-virus scanners.
++
+Beats required on hosts:
+
+* Winlogbeat
+windows_anomalous_script:: Searches for PowerShell scripts that potentially indicate
+malware execution or persistence mechanisms.
++
+PowerShell scripts with high levels of entropy (apparent randomness) point to the
```
Comment (randomuserid) [inline docs/en/siem/machine-learning.asciidoc]: actually it does not calculate entropy. The revised text is "The Windows anomalous script job looks for Powershell scripts with unusual data characteristics, such as obfuscation, that may be a characteristic of malicious powershell script text blocks."

File: docs/en/siem/machine-learning.asciidoc (hunk 1 above)
Comment (benskelker) [reply docs/en/siem/machine-learning.asciidoc]: @randomuserid 
How is this changed?

File: docs/en/siem/machine-learning.asciidoc (hunk 2 above)
Comment (benskelker) [reply docs/en/siem/machine-learning.asciidoc]: Corrected

File: docs/en/siem/machine-learning.asciidoc (hunk 3 above)
Comment (benskelker) [reply docs/en/siem/machine-learning.asciidoc]: corrected

File: docs/en/siem/machine-learning.asciidoc (hunk 4 above)
Comment (benskelker) [reply docs/en/siem/machine-learning.asciidoc]: corrected

File: docs/en/siem/machine-learning.asciidoc (hunk 1 above)
Comment (benskelker) [reply docs/en/siem/machine-learning.asciidoc]: Merging as is, we'll update this info afterwards.

**Pull request merged**